src/Credentials/ServiceAccountCredentials.php

```diff
@@ -164,7 +164,7 @@ public function __construct(
         ]);
 
         $this->projectId = $jsonKey['project_id'] ?? null;
-        $this->universeDomain = $jsonKey['universe_domain'] ?? null;
+        $this->universeDomain = $jsonKey['universe_domain'] ?? self::DEFAULT_UNIVERSE_DOMAIN;
```

Comment:
Should we also throw in case there's `domain-delegation`?
I am not very sure but it appears that we need to throw (near line#146) in case `sub` as well as `universe_domain` is set in the key, as per AL9 of [cloudsdk-tpc-authlibs](http://go/cloudsdk-tpc-authlibs#heading=h.y388tia22eux)

Reply:
Thank you for catching this, I completely missed it. PHP definitely supports `domain-delegation`

https://github.com/googleapis/google-auth-library-php/blob/b687d5efa8dc71782a5181c501adc28c4d89cdac/src/Credentials/ServiceAccountCredentials.php#L114-L115

I'll look more into it

Reply:
Okay, I added the exception and test in 5f8b409c2839b0a64e760af4faa8a050e783b47b, and added a fix for domain-wide delegation that we had missed in #505

Reply:
Approved it